Cosmetic/working with triggers

vebjornhaugland opened the pull request:
## Changes :balloon:

Added triggers to labels and entry fields. The labels at main page is now faded in to full opacity once they are displayed (by clicking the image).

The entry fields are now at 75% opacity, once they are in focus, they will display at 100% opacity. 

## Types of changes

What types of changes does your code introduce? _Mark boxes that apply with an `x`._

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist :heavy_check_mark:

_Mark boxes that apply with an `x`. These can also be filled out after the pull request is created._

- [ ] I have read the coding convention [guidelines](https://github.com/Dualog-students/dualog-students.github.io#company-guidelines) and my code follows these.
- [ ] Unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

<!-- The structure of your pull request can be altered if the template does not suit your needs -->

vebjornhaugland committed:
Start new branch clean and cool.

vebjornhaugland committed:
Make cosmetic changes to labels and entries with triggers

Add triggers to labels (text below logo once logo is clicked) on main page, so the labels are faded in as an animation. Entry fields on entry page view is faded when they are not in focus

IsakSundeSingh commented on Diplomatic/App.xaml:
```diff
@@ -15,6 +16,31 @@
+                        <local:OpacityTriggerAction StartsFrom="0"/>
+                    </Trigger.EnterActions>
+                    <Trigger.ExitActions>
+                        <local:OpacityTriggerAction StartsFrom="1"/>
```
I don't think this last exit-trigger is necessary. The label will be hidden before the trigger action starts, so the animation won't be visible.

IsakSundeSingh commented on Diplomatic/Utils/Triggers/OpacityAnimation.cs:
```diff
@@ -0,0 +1,25 @@
+        public int StartsFrom { set; get; } = 0;
+        //public double Opacity { get; set; } = 1;
+        //public uint Duration { get; set; } = 10000;
+
```
Commented code is not good, brah

IsakSundeSingh commented on Diplomatic/App.xaml:
```diff
@@ -15,6 +16,31 @@
         </Style>
 
+        <Style TargetType="Label">
+            <Setter Property="Opacity" Value="0.5"/>
```
Should labels have an opacity of 0.5? Entries have 0.75

vebjornhaugland committed:
Changes made based on constructive comments from Isak

Default opacity for labels is now 1. Remove commented code. Remove exit action for labels.

IsakSundeSingh commented on Diplomatic/Utils/Triggers/OpacityAnimation.cs:
```diff
@@ -0,0 +1,25 @@
+            visual.Animate("label", new Animation((d) => {
+                var val = StartsFrom == 0 ? d : 1 - d;
+                visual.Opacity = val;
+
```
Unnecessary newline.

IsakSundeSingh commented on Diplomatic/App.xaml:
```diff
@@ -17,16 +17,13 @@
 
         <Style TargetType="Label">
-            <Setter Property="Opacity" Value="0.5"/>
+            <Setter Property="Opacity" Value="1"/>
```
This can actually be removed, as the default opacity is 1.

IsakSundeSingh reviewed (approved):
👍
